### PR TITLE
[Tommy Collison] React-Todo

### PR DESCRIPTION
**Pull request by tommycollison**
📻 🔨 @John-Spraul 👋

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 ## Initializing the project.
 
-- `Fork and clone` this project and cd into your cloned version.
+- `Fork` and `clone` this project and cd into your cloned version.
```
Comment (John-Spraul) [inline README.md]: 🍴 & 👨👨

File: src/App.js [hunk 2]
```diff
@@ -1,16 +1,62 @@
-import React from 'react';
+import React, {Component} from 'react';
+import TodoList from './components/TodoComponents/TodoList';
+
+let storage = window.localStorage;
+let storageList = [];
+if (storage.list !== undefined){
+	storageList = JSON.parse(storage.list);
+}
```
Comment (John-Spraul) [inline src/App.js]: You're a strange one, Tommy. 🔥
I'd recommend not using local storage unless you're comfortable with the other material first

File: src/App.js [hunk 3]
```diff
@@ -1,16 +1,62 @@
...
+  this.state = {todo: storageList, current: ''}
+}
+
+handleInputChange = event => {
+  this.setState({current: event.target.value});
+}
+
+addTodo = (event, prevState) => {
+  event.preventDefault();
+  let clear = document.querySelector('.toDoForm input');
```
Comment (John-Spraul) [inline src/App.js]: Good job getting around whatever problem you were facing, but don't use querySelectors in React

File: src/App.js [hunk 4]
```diff
@@ -1,16 +1,62 @@
...
+  clear.value = '';
+  let newList = this.state.todo.slice();
+  let newItem = {task: this.state.current, id: Date.now(), completed: false};
```
Comment (John-Spraul) [inline src/App.js]: generally, it's best practice to lay out the object properties on new lines

File: src/App.js [hunk 5]
```diff
@@ -1,16 +1,62 @@
...
+  newList.push(newItem);
+  this.setState({todo: newList, current: ''});
+  storage.setItem('list', JSON.stringify(newList));
+}
+
+
+completeTask = event => {
+		let newList = this.state.todo.slice();
+		for (let todo in newList){
+			if (newList[todo].task === event.target.innerText && newList[todo].id === Number(event.target.dataset.id)){
```
Comment (John-Spraul) [inline src/App.js]: only need to check the `id`s

File: src/App.js [hunk 6]
```diff
@@ -1,16 +1,62 @@
...
+				newList[todo].completed = !newList[todo].completed;
```
Comment (John-Spraul) [inline src/App.js]: your function is called `completeTask`, but it looks like you're toggling the `completed` value, whether it be true or false

File: src/App.js [hunk 7]
```diff
@@ -1,16 +1,62 @@
...
+			}
+		}
+		this.setState({todo: newList});
+		storage.setItem('list', JSON.stringify(newList));
+	}
+
+removeCompleted = () => {
+  let currentList = this.state.todo.slice();
+  let newList = currentList.filter(todo => todo.completed === false);
```
Comment (John-Spraul) [inline src/App.js]: good use of filter (filter returns a new array, just FYI; meaning you don't need the slice)

File: src/App.js [hunk 8]
```diff
@@ -1,16 +1,62 @@
...
+  this.setState({todo: newList});
+  storage.setItem('list', JSON.stringify(newList));
+}
+
+
+render() {
+  return (
+    <div class="container">
+      <h1>Oh god another todo list app</h1>
+      <TodoList list={this.state.todo} onChange={this.handleInputChange} onSubmitButton={this.addTodo} onClear={this.removeCompleted} completeTask={this.completeTask} />
```
Comment (John-Spraul) [inline src/App.js]: might be best to break down this component onto multiple lines

File: src/components/TodoComponents/Todo.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+const Todo = (props) => {
+    return (
+        <div className="toDoContainer">
+            {
+            props.list.map(todo => <p className={`${todo.completed ? 'completed' : '' }`} onClick={props.completeTask} data-id={todo.id} key={todo.id}>{todo.task}</p>)
```
Comment (John-Spraul) [inline src/components/TodoComponents/Todo.js]: move this onto multiple lines

File: src/components/TodoComponents/TodoForm.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'; 
+
+const TodoForm = (props) => {
+    return (
+      <form onSubmit={props.onSubmitButton} className="toDoForm">
```
Comment (John-Spraul) [inline src/components/TodoComponents/TodoForm.js]: good use of a form

File: src/components/TodoComponents/TodoList.js [hunk 11]
```diff
@@ -1,2 +1,17 @@
...
 // feel free to change this component.js into TodoList.js
+
+import React from 'react';
+import Todo from './Todo';
+import TodoForm from './TodoForm';
+
+const TodoList = (props) => {
+    return(
+        <div className="toDoListContainer">
+        <TodoForm onChange={props.onChange} onSubmitButton={props.onSubmitButton} onClear={props.onClear} />
```
Comment (John-Spraul) [inline src/components/TodoComponents/TodoList.js]: instructions ask for the form to be on the `App.js` component 🤷‍♂️

**Comment by tommycollison** [comment]
Okay! 

1. refactor day 1 code so I have something clean to work with 
2. do day 2 MVP requirements
3. hopefully get to day 2 stretchy mcstretch goals
4. ???
5. profit!

File: src/App.js [hunk 12]
```diff
@@ -1,62 +1,69 @@
...
-  this.setState({current: event.target.value});
-}
+  constructor(props){
+    super(props);
+    this.state={
+      input: '',
+     
+      todolist: [
+        {
+          task: 'graduate lambda school',
```
Comment (John-Spraul) [inline src/App.js]: ✔️

File: src/App.js [hunk 13]
```diff
@@ -1,62 +1,69 @@
...
-  this.setState({todo: newList, current: ''});
-  storage.setItem('list', JSON.stringify(newList));
-}
+      ],
+     
+    };
+  }
 
+  handleInput = (event) => {
+    const {value} = event.target;
```
Comment (John-Spraul) [inline src/App.js]: do you know what this line is doing?

File: src/App.js [hunk 14]
```diff
@@ -1,62 +1,69 @@
...
-removeCompleted = () => {
-  let currentList = this.state.todo.slice();
-  let newList = currentList.filter(todo => todo.completed === false);
-  this.setState({todo: newList});
-  storage.setItem('list', JSON.stringify(newList));
-}
+  handleClick = (id) => {
+    let list = this.state.todolist.slice();
+    let todo = list.findIndex(todo => todo.id === id);
+    list[todo].completed = true
```
Comment (John-Spraul) [inline src/App.js]: you could change this to toggle between true and false with `list[todo].completed = !list[todo].completed`

File: src/components/TodoComponents/Todo.js [hunk 15]
```diff
@@ -1,14 +1,13 @@
...
-            {
-            props.list.map(todo => <p className={`${todo.completed ? 'completed' : '' }`} onClick={props.completeTask} data-id={todo.id} key={todo.id}>{todo.task}</p>)
-            }
-        </div>
-    );
-}
 
+ const Todo = (props) => {
+     return(
+         <li className={props.value.completed ? "completed" : null} onClick={() => props.handleClick(props.value.id)}>
```
Comment (John-Spraul) [inline src/components/TodoComponents/Todo.js]: Remember the struggle 😣

File: src/components/TodoComponents/TodoForm.js [hunk 16]
```diff
@@ -1,14 +1,13 @@
...
-      </form>
-    );
-  }
-  export default TodoForm;
\ No newline at end of file
+        <form>
+            <input value={props.input} onChange={props.handleInput} />
+            <button onClick={props.handleSubmit}>Add</button>
+            <button>Clear</button>
+        </form>
```
Comment (John-Spraul) [inline src/components/TodoComponents/TodoForm.js]: Does this form work 100% as you expect it to?

**Pull request closed without merging**